File: FarNetCSharp/Class1.cs
﻿using System;
using FarNet;

[System.Runtime.InteropServices.Guid("f584e496-b801-4df3-be4f-2848d3a6e77c")]
[ModuleTool(Name = "Hello.C#", Options = ModuleToolOptions.AllMenus)]
public class Hello : ModuleTool
{
	public override void Invoke(object sender, ModuleToolEventArgs e)
	{
		Far.Api.Message("Hello from C# module!");
	}
}

File: FarNetFSharp/Class1.fs
﻿module Test
open FarNet

[<ModuleTool(Name = "Hello.F#", Options = ModuleToolOptions.AllMenus)>]
[<Guid("cc0ecb2f-552a-4fe4-90f0-227077386890")>]
type FSharpTool () =
    inherit ModuleTool ()
    override __.Invoke (_, e) =
        Far.Api.Message "Hello from F# module!"
